### PR TITLE
EVG-2329: avoid terminating hosts while waiting for agents

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: monitor/host_flagging.go [hunk 1]
```diff
@@ -114,14 +113,26 @@ func flagIdleHosts(d []distro.Distro, s *evergreen.Settings) ([]host.Host, error
...
-		//  idle for at least 15 minutes or last communication time has been more than 10 mins and
-		//  less than 5 minutes til next payment
-		if (communicationTime >= CommunicationTimeCutoff || idleTime >= IdleTimeCutoff) &&
-			tilNextPayment <= MaxTimeTilNextPayment {
-			idleHosts = append(idleHosts, freeHost)
+		if tilNextPayment > MaxTimeTilNextPayment {
+			continue
+		}
+
+		if freeHost.IsWaitingForAgent() && (communicationTime >= idleWaitingForAgentCutoff || idleTime >= idleWaitingForAgentCutoff) {
```
Comment (john-m-liu) [inline monitor/host_flagging.go]: this should be <=

File: monitor/host_flagging.go [hunk 2]
```diff
@@ -114,14 +113,26 @@ func flagIdleHosts(d []distro.Distro, s *evergreen.Settings) ([]host.Host, error
...
+			grip.Warning(message.Fields{
```
Comment (john-m-liu) [inline monitor/host_flagging.go]: This should probably be debug level

File: model/host/host.go [hunk 3]
```diff
@@ -392,6 +392,20 @@ func (h *Host) SetAgentRevision(agentRevision string) error {
 	return nil
 }
 
+// IsWaitingForAgent provides a local predicate for the logic in the
+// "NeedsNewAgent" query.
+func (h *Host) IsWaitingForAgent() bool {
```
Comment (john-m-liu) [inline model/host/host.go]: Should this handle if the LCT field is 0?

**Comment by tychoish** [comment]
updated!

**Pull request merged**